src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -996,20 +815,11 @@ void MachEpilogNode::emit(CodeBuffer& cbuf, PhaseRegAlloc* ra_) const
   // Note that VerifyStackAtCalls' Majik cookie does not change the frame size popped here
 
   if (framesize) {
-    emit_opcode(cbuf, Assembler::REX_W);
-    if (framesize < 0x80) {
-      emit_opcode(cbuf, 0x83); // addq rsp, #framesize
-      emit_rm(cbuf, 0x3, 0x00, RSP_enc);
-      emit_d8(cbuf, framesize);
-    } else {
-      emit_opcode(cbuf, 0x81); // addq rsp, #framesize
-      emit_rm(cbuf, 0x3, 0x00, RSP_enc);
-      emit_d32(cbuf, framesize);
-    }
+    __ addq(as_Register(RSP_enc), framesize);
```

Comment:
Isn't this changing the encoding of the add?

Reply:
Thanks for reviewing @TobiHartmann . Why do you think it changes the encoding?

Reply:
Because before your change, the add is encoded with `0x83` if the framesize is small enough and I think `Assembler::addq` always uses the `0x81` encoding, right?

Reply:
It's kind of confusing but the encoding will end up being the same. The `addq` being called here is the one at assembler_x86.cpp:12899. In that method, the `prefixq_and_encode` will emit the REX_W prefix because the encoding of RSP is < 8. The `emith_arith`, defined on line 313 of the same file, will check that the immediate passed as argument (i.e., framesize) fits in a byte, if so it'll actually emit "0x81 | 0x02".

Reply:
Ah, perfect. Thanks for the clarification!